packages/shared/routes/dashboard/wallet/views/AccountHistory.svelte

```diff
@@ -72,6 +80,41 @@
             }
         }
     }
+
+    const filters = ['all', 'incoming', 'outgoing']
+    let activeFilterIndex = 0
+
+    let searchActive = false
+    $: searchValue = searchActive ? searchValue : ''
+
+    let filteredTransactions = transactions
+    $: switch (activeFilterIndex) {
+        case 0:
+            filteredTransactions = transactions
+            break
+        case 1:
+            filteredTransactions = transactions.filter(transaction => getIncomingFlag(transaction?.payload))
+            break
+        case 2:
+            filteredTransactions = transactions.filter(transaction => !getIncomingFlag(transaction?.payload))
+            break
+        default:
+            filteredTransactions = transactions
+    }
+
+    let queryTransactions = filteredTransactions
+    $: if (searchValue) {
+        queryTransactions = filteredTransactions.filter(transaction => {
+            if (transaction?.payload?.type === 'Transaction') {
+                return transaction?.payload?.data?.essence?.data?.value?.toString()?.includes(searchValue) ||
+                    sendAddressFromTransactionPayload(transaction?.payload).includes(searchValue) ||
+                    receiverAddressesFromTransactionPayload(transaction?.payload).find(addr => addr.includes(searchValue))
+            }
+            return transaction?.id?.includes(searchValue)
```

Comment:
This will only search the messageID if the transaction is not of type `Transaction`